pkg/reconciler/serverlessservice/serverlessservice.go

```diff
@@ -162,10 +162,10 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 
 	// Don't modify the informers copy.
 	sks := original.DeepCopy()
-	err = r.reconcile(ctx, sks)
-	if err != nil {
-		r.Recorder.Eventf(sks, corev1.EventTypeWarning, "UpdateFailed", "InternalError: %v", err.Error())
-		return err
+	reconcileErr := r.reconcile(ctx, sks)
+	if reconcileErr != nil {
+		r.Recorder.Eventf(sks, corev1.EventTypeWarning, "UpdateFailed", "InternalError: %v", reconcileErr.Error())
+		return reconcileErr
```

Comment:
This reconciler has a different pattern than the others. All the others will attempt to update the status, before returning the `reconcileErr`. This one doesn't update status if `reconcileErr` is not `nil`. Is there a reason for the difference? Should I change it to match the pattern of the others?

Reply:
@vagababov

Reply:
As discussed.